### ISSUE TITLE
Charsches after round about 220 cards

### ISSUE THREAD

**Issue by Alyxvance** [ankidroid/Anki-Android#5858]
###### Reproduction Steps

1. deck with round 8000 cards
2. card settings 500 oder more to leran 
3. 
###### Actual Result
anki crashes after round 220 cards you have leraned 
without response back to the deck menue

###### Debug info
AnkiDroid Version = 2.9.5
Android Version = 10
ACRA UUID = 6013f375-1754-4ba8-85c5-696fdd57b697

###### Research
*Enter an [ x ] character to confirm the points below:*

[x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid

[x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue

[x] I have searched for similar existing issues here and on the user forum
if its a duplicate then sorry

**Comment by david-allison-1** [comment]
Hi @Alyxvance ,

Thank you so much for the bug report, and sorry for the inconvenience. We use Chrome to draw your flash cards, and large fonts can cause it to run out of memory and crash. 

We can't fix this directly, but I've put in a fix to AnkiDroid last week to improve the user experience (#5780). You can get this now by [joining our beta program and downloading 2.9.6beta1](https://docs.ankidroid.org/manual.html#betaTesting).

If you find any other issues, or this doesn't solve your problem, please let us know (via the report crash dialog, and on here). We're on a bug quashing spree right now, and any help in tracking issues is invaluable.

Thanks for using AnkiDroid,
David

----

Also related: #5820, #4967.  
#4856 discusses alternate fonts to use.

**Comment by Alyxvance** [comment]
thanx for you response
my font is only 7mb, thats waht i not understand cause with out font works
and i can't see thats my memory run out i dont know how i can see it live
i have round about 400 mb memory usage every time i locked for   
i have had the 2.9.6beta3 hotfix compiled and aktual the master brunch 2.10alpha54
both with some changes 
i have implementy this 
https://github.com/ankidroid/Anki-Android/issues/4841
and 
https://github.com/ankidroid/Anki-Android/commit/0731
i have just installet the beta from playstore 
AnkiDroid Version = 2.9.6beta3
Android Version = 10
ACRA UUID = b7cf581b-b92b-4604-98fe-b7c14d2574d3
Mate 20 Pro Android 10.0.0.188
all have the same issue 
the next waht i wonder is i do not have chrome on my phone this was the first i have deleted per ADB ... and replaced with a chrome clone (Opera)
well, i mean its a hard thing 200 cars to review not sure. the most peps will not realy review 200 cards

**Comment by david-allison-1** [comment]
> thanx for you response
> my font is only 7mb, thats waht i not understand cause with out font works
> and i can't see thats my memory run out i dont know how i can see it live
> i have round about 400 mb memory usage every time i locked for
> i have had the 2.9.6beta3 hotfix compiled and aktual the master brunch 2.10alpha54
> both with some changes
> i have implementy this
> #4841
> and
> [0731](https://github.com/ankidroid/Anki-Android/commit/0731)
> i have just installet the beta from playstore
> AnkiDroid Version = 2.9.6beta3
> Android Version = 10
> ACRA UUID = b7cf581b-b92b-4604-98fe-b7c14d2574d3
> Mate 20 Pro Android 10.0.0.188
> all have the same issue
> the next waht i wonder is i do not have chrome on my phone this was the first i have deleted per ADB ... and replaced with a chrome clone (Opera)
> well, i mean its a hard thing 200 cars to review not sure. the most peps will not realy review 200 cards

Thanks, I expect it to b fixed, but give me an @ if it still occurs. 

It's actually the System WebView, which I believe is still Chromium based, rather than Chrome itself. I'm not too familiar with that area of Android, but I believe it's near mandatory for apps to have and access. Uninstalling Chrome shouldn't cause an issue.

All the best,
David

**Comment by mikehardy** [comment]
It is possible to swap in 'WebView' compatible implementations as the default webview and we've solved bugs before for people that wanted to use Tor even (#5337). 200 cards should be possible. The patch before (present in 2.9.6beta3 and above) should be handling it. We'll definitely need debug information if we're going to track this down. 7Mb as a font is large - I believe people have unreasonable expectations of mobile technology these days as so many amazing things are possible ;-). We can try to work around this, but the most reliable way to fix it is via self-service where you optimize your font usage somehow so it fits within your device's capabilities

**Comment by Alyxvance** [comment]
Ähm Well i Not Sure wahts acra is, or have i some to enable?
How can i give you a Log? I have had installed the Beta Version new cause i have Made little Changes See above. Btw 7mb for a ckj Font is mit big ... But it shoult Not be ... A Font is usualy ontime loded ;)
An yes Anki says thats the webviewe Crashes ...

**Comment by mikehardy** [comment]
https://docs.ankidroid.org/help.html - "Bug Reports and Feature Requests"

**Comment by Alyxvance** [comment]
Sorry but I don't understand you . I have read this page 10 times. But more then 
"AnkiDroid Version = 2.9.6beta3
Android Version = 10
ACRA UUID = b7cf581b-b92b-4604-98fe-b7c14d2574d3"
I do not get don't know if the Beta or aloha sends acralyzer data ... And u don't find any crash report ... I will so help you but u can't with out dates ...

**Comment by mikehardy** [comment]
Well, I still don't see any crashes for the ACRA UUID, but I think that is expected now as we quietly work around the problem. If you are seeing the toast about the WebView crashing, that's basically the problem and there isn't a lot we can do other than to keep working around it (which should be acceptable to users - it is almost as if nothing happened and you just keep reviewing)

Here is where we do that workaround now, in 2.9.6beta3 and higher or the current alphas: https://github.com/ankidroid/Anki-Android/blob/master/AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java#L1641 - nothing is logged anymore, we just workaround, toast the info to the user and keep on moving

**Comment by Alyxvance** [comment]
okay,
maybe not a realy problem of ankidroid  more of googles webviewer
well the redraw works its not a full crash ... i'm not sure but i think sometimes gets only the message webviewer then is okay and go on ... not sure i'll see ...
the old 2.8 version has a full crash 

solong
thank you

**Comment by mikehardy** [comment]
closing preliminarily because I think it's fixed by #5820 - we can re-open if it's still blowing up though of course

**Comment by Alyxvance** [comment]
AnkiDroid Version = 2.15.6
Android Version = 10
Manufacturer = HUAWEI
Model = LYA-L29
Hardware = kirin980

Webview User Agent = Mozilla/5.0 (Linux; Android 10; LYA-L29 Build/HUAWEILYA-L29; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/109.0.5414.85 Mobile Safari/537.36

ACRA UUID = e686103d-2874-4a73-9c9a-9e9d430a889c
Scheduler = std2
Crash Reports Enabled = true
DatabaseV2 Enabled = true

Hi Not sure If I right Here cause is long time after 
But the webviewer still crashes after some cards 
I got as font Kanjistroke Order in. 
Its more info thats the issue still Up 
atm ankidroid reloads the last card after crash.
I got used to it. 
Well 
Greetings
